lib/modules/manager/mint/extract.spec.ts

```diff
@@ -0,0 +1,113 @@
+yonaskolb/xcodegen
+#yonaskolb/xcodegen
+#realm/SwiftLint@0.48.0#commented out
+`;
+
+describe('modules/manager/mint/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('Makefile With Version Description', () => {
+      const res = extractPackageFile(simpleMintfile);
+      expect(res).toEqual({
+        deps: [
+          {
+            depName: 'SwiftGen/SwiftGen',
+            currentValue: '6.6.1',
+            datasource: 'git-tags',
+            packageName: 'https://github.com/SwiftGen/SwiftGen.git',
+          },
+          {
+            depName: 'yonaskolb/xcodegen',
+            currentValue: '2.30.0',
+            datasource: 'git-tags',
+            packageName: 'https://github.com/yonaskolb/xcodegen.git',
+          },
+          {
+            depName: 'realm/SwiftLint',
+            currentValue: '0.48.0',
+            datasource: 'git-tags',
+            packageName: 'https://github.com/realm/SwiftLint.git',
+          },
+        ],
+      });
+    });
+
+    it('Makefile Without Version Description', () => {
+      const res = extractPackageFile(noVersionMintfile);
+      expect(res).toEqual({
+        deps: [
+          {
+            depName: 'yonaskolb/xcodegen',
+            currentValue: null,
+            skipReason: 'no-version',
+            packageName: 'https://github.com/yonaskolb/xcodegen.git',
+          },
+          {
+            depName: 'realm/SwiftLint',
+            currentValue: null,
+            skipReason: 'no-version',
+            packageName: 'https://github.com/realm/SwiftLint.git',
+          },
+        ],
+      });
+    });
+
+    it('Complex Makefile', () => {
+      const res = extractPackageFile(complexMintFileContent);
+      expect(res).toEqual({
+        deps: [
+          {
+            depName: 'SwiftGen/SwiftGen',
+            currentValue: '6.6.1',
+            datasource: 'git-tags',
+            packageName: 'https://github.com/SwiftGen/SwiftGen.git',
+          },
+          {
+            depName: 'yonaskolb/xcodegen',
+            currentValue: null,
+            skipReason: 'no-version',
+            packageName: 'https://github.com/yonaskolb/xcodegen.git',
+          },
+          {
+            depName: 'realm/SwiftLint',
+            currentValue: '0.48.0',
+            datasource: 'git-tags',
+            packageName: 'https://github.com/realm/SwiftLint.git',
+          },
+        ],
+      });
+    });
+
+    it('Mintfile Includes Commented Out', () => {
+      const res = extractPackageFile(includesCommentOutMintFileContent);
+      expect(res).toEqual({
+        deps: [
+          {
+            depName: 'SwiftGen/SwiftGen',
+            currentValue: '6.6.1',
+            datasource: 'git-tags',
+            packageName: 'https://github.com/SwiftGen/SwiftGen.git',
+          },
+          {
+            currentValue: null,
+            depName: 'yonaskolb/xcodegen',
+            packageName: 'https://github.com/yonaskolb/xcodegen.git',
```

Comment:
why only the first has datasource?

Reply:
Sorry!🙇‍♂️ I forgot to add!
I add in https://github.com/renovatebot/renovate/pull/17351/commits/8d465c705df36b735525e659fbcd7cf9252f9273